src/responses/search-response-details.ts

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { Aggregation } from '../models/aggregation';
+import { Hit, HitFactory } from '../models/hit-types/hit';
+import type { SearchHitSchema } from './search-hit-schema';
+
+export interface SearchResponseBody {
+  hits: SearchResponseHits;
+  aggregations?: Record<string, Aggregation>;
+}
+
+export interface SearchResponseHits {
+  total: number;
+  returned: number;
+  hits: Record<string, any>[];
+}
+
+/**
+ * This is the search response details inside the SearchResponse object that contains
+ * the search results, under the `docs` key.
+ *
+ * @export
+ */
+export class SearchResponseDetails {
+  /**
+   * Total number of results found
+   */
+  totalHits: number;
+
+  /**
+   * Number of returned hits in this response
+   */
+  returnedHits: number;
+
+  /**
+   * The array of search results
+   */
+  hits: Hit[];
+
+  /**
+   * Requested aggregations such as facets or histogram data
+   *
+   * @type {Record<string, Aggregation>}
+   * @memberof SearchResponseDetails
+   */
+  aggregations?: Record<string, Aggregation>;
+
+  /**
+   * The hit schema for this response
+   */
+  schema?: SearchHitSchema;
+
+  constructor(body: SearchResponseBody, schema: SearchHitSchema) {
+    this.schema = schema;
+    const hitType = schema?.hit_type;
+
+    this.totalHits = body?.hits?.total ?? 0;
+    this.returnedHits = body?.hits?.returned ?? 0;
+    this.hits =
+      body?.hits?.hits?.map((hit: Hit) =>
+        HitFactory.createFromType(hitType, hit)
```

Comment:
```suggestion
        this.formatResult(hitType, hit)
```
 something like that?
 
`hits` feels less descriptive than `response` ... idk.. just thinking out loud

Reply:
Agreed... `results` instead of `hits` is probably clearer. (Notably, I think the old service used `docs` for this, which maybe suffers from the same vagueness as `hits` 😄)